### PR TITLE
docs(contact-center): add ai-docs SDD pattern documentation

### PR DESCRIPTION
**Pull request by ciscoRankush**
# COMPLETES #SPARK-775699

  ## This pull request addresses

  AI agents need reference documentation for the established coding patterns in the @webex/contact-center SDK so they can generate code that is consistent with existing conventions. Without pattern docs, agents
  produce code that uses console.log instead of LoggerProxy, misses MetricsManager tracking, uses incorrect error handling, and deviates from established type conventions.

  ## by making the following changes

  - Adds `ai-docs/patterns/` with 5 pattern documentation files:
    - **typescript-patterns.md** — Type conventions, LoggerProxy usage (not console.log), error handling with getErrorDetails, constant patterns, JSDoc conventions
    - **testing-patterns.md** — Jest test structure, mocking patterns (LoggerProxy, services, WebSocket), test naming conventions, success/error/edge case coverage
    - **event-driven-patterns.md** — Event emission patterns, listener registration/cleanup, event constant usage, WebSocket event handling
    - **websocket-patterns.md** — WebSocket connection lifecycle, reconnection flow, silent relogin, message handling patterns
    - **sdk-plugin-patterns.md** — WebexPlugin extension, service registration, initialization in READY block, public API exposure via cc.ts

  ### Change Type

  - [ ] Bug fix (non-breaking change which fixes an issue)
  - [ ] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to change)
  - [x] Documentation update
  - [ ] Tooling change
  - [ ] Internal code refactor

  ## The following scenarios were tested

  1. **Pattern Compliance** — Verified all code examples in pattern files follow SDK conventions (LoggerProxy, not console.log; getErrorDetails for errors; MetricsManager tracking)
  2. **console.log Fix** — Confirmed typescript-patterns.md uses LoggerProxy.error() in error handling examples (previously had console.log which contradicted the no-console.log rule)
  3. **Cross-Reference Validation** — Verified pattern files are correctly referenced from template 00-master.md files via `../../patterns/` relative paths
  4. **Type E Routing** — Tested that architecture understanding prompts (e.g., "How does WebSocket reconnection work?") correctly route to these pattern docs via the Service Routing Table

  ## The GAI Coding Policy And Copyright Annotation Best Practices ##

  - [ ] GAI was not used (or, no additional notation is required)
  - [ ] Code was generated entirely by GAI
  - [x] GAI was used to create a draft that was subsequently customized or modified
  - [ ] Coder created a draft manually that was non-substantively modified by GAI (e.g., refactoring was performed by GAI on manually written code)
  - [x] Tool used for AI assistance (GitHub Copilot / Other - specify)
    - [ ] Github Copilot
    - [x] Other - Claude Code (Claude Opus 4.6)
  - [x] This PR is related to
    - [ ] Feature
    - [ ] Defect fix
    - [x] Tech Debt
    - [ ] Automation

  ### I certified that

  - [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
  - [x] I discussed changes with code owners prior to submitting this pull request
  - [x] I have not skipped any automated checks
  - [x] All existing and new tests passed
  - [x] I have updated the documentation accordingly

### CODE REVIEW

**Comment by Shreyas281299** [comment]
@codex review

File: packages/@webex/contact-center/ai-docs/patterns/sdk-plugin-patterns.md
Comment (Shreyas281299) [inline packages/@webex/contact-center/ai-docs/patterns/sdk-plugin-patterns.md:1]: We can remove this

File: packages/@webex/contact-center/ai-docs/patterns/typescript-patterns.md [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+// Step 2: Create union type from values
+type Enum<T extends Record<string, unknown>> = T[keyof T];
+export type CC_EVENTS = Enum<typeof CC_EVENTS>;
+
+// Usage:
+const event: CC_EVENTS = CC_EVENTS.AGENT_LOGIN; // ✅ Type-safe
+const event2: CC_EVENTS = 'InvalidEvent';        // ❌ Compile error
+```
+
+### Response Type Pattern
```
Comment (Shreyas281299) [inline packages/@webex/contact-center/ai-docs/patterns/typescript-patterns.md]: Lets remove this

File: packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+      logger: {
+        log: jest.fn(),
+        error: jest.fn(),
+        info: jest.fn(),
+      },
+      credentials: {
+        getOrgId: jest.fn(() => 'mockOrgId'),
+      },
+      config: config,
+    }) as unknown as WebexSDK;
```
Comment (Shreyas281299) [inline packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md]: We should not use unkown, thie way LLM would put unknown everywhere

File: packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+
+---
+
+## Test Coverage Goals
+
+Target: **85% coverage**
+
+```bash
+# Run tests with coverage
+yarn workspace @webex/contact-center test --coverage
```
Comment (Shreyas281299) [inline packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md]:
```suggestion
yarn workspace @webex/contact-center test:unit --coverage
```

File: packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+```
+
+---
+
+## WebSocket Event Flow
+
+### Message Reception
+
+```
+WebSocket → WebSocketManager → cc.handleWebsocketMessage → Event Emission
```
Comment (Shreyas281299) [inline packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md]: Mermaid diagram. This is a very speicifc flow. We can add a generic flow to explain the generic event reception

File: packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+  }
+  
+  public handleIncomingTask(taskData: TaskData) {
+    const task = this.createTask(taskData);
+    this.emit(TASK_EVENTS.TASK_INCOMING, task);
+  }
+}
+```
+
+---
```
Comment (Shreyas281299) [inline packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md]: - We dont need subheading

What we need?
1. How we emit events. Two ways
2. We need to differntiate when to use trigger and when to use emit
P.S: emit is used when the class is extendin EventEmitter and trigger is used in the other case-trigger is very speicifc for cc object as it extends WebexPlugin
3. Code example of each way

File: packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+  });
+}
+
+// Cleanup in deregister
+public async deregister() {
+  this.taskManager.off(TASK_EVENTS.TASK_INCOMING, this.handleIncomingTask);
+  this.services.webSocketManager.off('message', this.handleWebsocketMessage);
+}
+```
+
```
Comment (Shreyas281299) [inline packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md]: What we need?
1. How to add/remove an event listener?
2. Examples
    2.1  Internal event listening
    2.2  Events sent to application (from cc,task)

File: packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+      module: CC_FILE,
+      method: 'handleConnectionLost',
+    });
+    
+    if (this.$config?.allowAutomatedRelogin) {
+      await this.silentRelogin();
+    }
+  }
+}
+```
```
Comment (Shreyas281299) [inline packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md]: This is not part of SDK. Lets remove this

File: packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md [hunk 8]
```diff
@@ -0,0 +1,464 @@
...
+  // Act & Assert
+  await expect(
+    webex.cc.stationLogin({ teamId: 'team-1', loginOption: 'BROWSER' })
+  ).rejects.toThrow('INVALID_CREDENTIALS');
+});
+```
+
+---
+
+## Event Testing
```
Comment (Kesari3008) [inline packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md:289]: Here I believe we need few tweaks, event listeners and their callback invocations are tested by spying on it and fetch the callback and invoking it using mock.calls. So phrasing this section like that would be helpful

File: packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md [hunk 9]
```diff
@@ -0,0 +1,464 @@
...
+  });
+
+  // Act
+  const result = await webex.cc.stationLogin({
+    teamId: 'team-1',
+    loginOption: 'BROWSER',
+  });
+
+  // Assert
+  expect(result).toEqual(expect.objectContaining({
```
Comment (Kesari3008) [inline packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md]: Also I would not encourage using expect.objectContaiining in tests. We prefer to do exact matches as much as possible in SDK unit testing

File: packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md [hunk 10]
```diff
@@ -0,0 +1,464 @@
...
+### Test File Template
+
+```typescript
+import 'jsdom-global/register';
+import MockWebex from '@webex/test-helper-mock-webex';
+import ContactCenter from '../../../src/cc';
+import {WebexSDK} from '../../../src/types';
+import config from '../../../src/config';
+
+// Mock dependencies
```
Comment (Kesari3008) [inline packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md:53]: We already have a section talking about mocking logger proxy so why are we adding this here as well ?

File: packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md [hunk 11]
```diff
@@ -0,0 +1,464 @@
+# Testing Patterns - Contact Center SDK
+
+> **Purpose**: Jest testing patterns and conventions for the Contact Center SDK.
+
+---
+
+## Test Structure
```
Comment (Kesari3008) [inline packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md:7]: We should add information like for every new file created we need to create a corresponding test file and it's location should be decided based on below file location structure attached

File: packages/@webex/contact-center/AGENTS.md [hunk 12]
```diff
@@ -0,0 +1,658 @@
...
+
+**Do not guess. Do not default to the most common type. Ask.**
+
+---
+
+## Task Type Routing
+
+**A. Create New Service**
+- Use when adding a new service module (for example AddressBook/Queue/EntryPoint style additions).
+- **Route to:** [`ai-docs/templates/new-service/00-master.md`](ai-docs/templates/new-service/00-master.md)
```
[the remark this answers is not in the document]
Comment (ciscoRankush) [reply packages/@webex/contact-center/AGENTS.md:87]: Templates are available as part of different PRs #4722

File: packages/@webex/contact-center/AGENTS.md [hunk 13]
```diff
@@ -0,0 +1,658 @@
...
+
+For deep understanding of specific services, read their `ai-docs/`. When working on a task that affects a specific service, **always load that service's AGENTS.md and ARCHITECTURE.md** for scope-specific guidance.
+
+### Service Routing Table
+
+Use this table to identify which service's ai-docs to load based on the developer's task:
+
+| Service | Scope / Keywords | AGENTS.md | ARCHITECTURE.md |
+|---------|-----------------|-----------|-----------------|
+| **Agent** | login, logout, state change, buddy agents, station | [`src/services/agent/ai-docs/AGENTS.md`](src/services/agent/ai-docs/AGENTS.md) | [`src/services/agent/ai-docs/ARCHITECTURE.md`](src/services/agent/ai-docs/ARCHITECTURE.md) |
```
[the remark this answers is not in the document]
Comment (ciscoRankush) [reply packages/@webex/contact-center/AGENTS.md:423]: Templates are available as part of different PRs , and will be raised by other developers . We have just provided placeholders here

File: packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md (hunk 11 above)
Comment (ciscoRankush) [reply packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md:7]: Done

File: packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md (hunk 10 above)
Comment (ciscoRankush) [reply packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md:53]: Removed the ## Mocking LoggerProxy section

File: packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md (hunk 8 above)
Comment (ciscoRankush) [reply packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md:289]: Done

File: packages/@webex/contact-center/ai-docs/patterns/typescript-patterns.md (hunk 1 above)
Comment (ciscoRankush) [reply packages/@webex/contact-center/ai-docs/patterns/typescript-patterns.md]: Done

File: packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md (hunk 2 above)
Comment (ciscoRankush) [reply packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md]: Done

File: packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md (hunk 3 above)
Comment (ciscoRankush) [reply packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md]: Done

File: packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md (hunk 5 above)
Comment (ciscoRankush) [reply packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md]: Done

File: packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md (hunk 6 above)
Comment (ciscoRankush) [reply packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md]: Done

File: packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md (hunk 7 above)
Comment (ciscoRankush) [reply packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md]: Done

File: packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md (hunk 9 above)
Comment (ciscoRankush) [reply packages/@webex/contact-center/ai-docs/patterns/testing-patterns.md]: Done

**Review by Shreyas281299 — APPROVED** [review]

File: packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md (hunk 4 above)
Comment (ciscoRankush) [reply packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md]: this is more machine readable statement , so lets keep it

File: packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md [hunk 14]
```diff
@@ -0,0 +1,450 @@
...
+    // ... more cases
+  }
+};
+```
+
+---
+
+## Event Emission
+
+There are **two methods** for emitting events: `trigger` and `emit`. Both work on the `cc` object (which extends `WebexPlugin`), but they should be used in different contexts.
```
Comment (Kesari3008) [inline packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md]: emit doesn't work on cc object because it doesn't extend event emitter class. We are adding @ts-ignore to be able to invoke it. Emphasis here should be that classes which extend EventEmitter can use emit for event emission but for classes like cc.ts can use trigger for emission

File: packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md [hunk 15]
```diff
@@ -0,0 +1,450 @@
...
+// Add a listener
+source.on(EVENT_CONSTANT, handler);
+
+// Remove a listener (must pass the same function reference)
+source.off(EVENT_CONSTANT, handler);
+```
+
+Always store handler references as **arrow function properties** so you can remove them later:
+
+```typescript
```
Comment (Kesari3008) [inline packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md:220]: We should speciifcally add a subheading here saying example

File: packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md [hunk 16]
```diff
@@ -0,0 +1,450 @@
...
+
+### Events Sent to Application (from cc, task)
+
+Application consumers subscribe to events on the `cc` object or on task instances:
+
+```typescript
+const cc = webex.cc;
+
+// Agent events (from cc)
+cc.on(AGENT_EVENTS.AGENT_STATE_CHANGE, (event) => {
```
Comment (Kesari3008) [inline packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md]: In these examples, we should have callbacks while listening because .off won;t work if we don;t have separate callback function

**Review by Kesari3008 — APPROVED** [review]

File: packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md [hunk 17]
```diff
@@ -0,0 +1,485 @@
...
+```typescript
+// cc.ts — ContactCenter extends WebexPlugin
+// @ts-ignore
+this.trigger(TASK_EVENTS.TASK_INCOMING, task);
+
+// @ts-ignore
+this.trigger(TASK_EVENTS.TASK_HYDRATE, task);
+```
+
+> **Note**: `trigger` requires `// @ts-ignore` because WebexPlugin's TypeScript type definitions don't expose it.
```
Comment (Kesari3008) [inline packages/@webex/contact-center/ai-docs/patterns/event-driven-patterns.md:167]: This should be emit here, it should say that we can still use emit but with @ts-ignore. We can address this PRs which are not merged yet
